client/src/main/java/de/hellfish/ngrok/client/NgrokClient.java

```
@@ -0,0 +1,132 @@
+package de.hellfish.ngrok.client;
+
+import de.hellfish.ngrok.utils.HttpRequest;
+import de.hellfish.ngrok.utils.HttpRequestSerializer;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+import java.net.UnknownHostException;
+
+@Slf4j
+@Setter
+@Getter
+public class NgrokClient implements Runnable {
+
+    private int servicePort; // 8082
+    private String serviceProtocol;
+    private static final String SERVER_HOST = "localhost";
+    private static final int SERVER_PORT = 8082;
+    private Socket serverSocket;
+    private PrintWriter serverOut;
+    private BufferedReader serverIn;
+    private String linkFromServer;
+    private boolean running;
+
+    public NgrokClient(int servicePort, String serviceProtocol) {
+        this.servicePort = servicePort;
+        this.serviceProtocol = serviceProtocol;
+    }
+
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            log.error("Provide service protocol and port. Syntax: java NgrokClient <protocol> <port>");
+            return;
+        }
+        String serviceProtocol = args[0];
+        int servicePort = 0;
+        try {
+            servicePort = Integer.parseInt(args[1]);
+        } catch (NumberFormatException ignored) {
```

Never ignore exceptions on user input.

---

Fixed